webofneeds/won-node/src/main/java/won/node/camel/predicate/IsSystemMessageToOwnerPredicate.java

```diff
@@ -32,7 +32,12 @@
     public boolean matches(Exchange exchange) {
         WonMessage message = (WonMessage) exchange.getIn().getHeader(WonCamelConstants.ORIGINAL_MESSAGE_HEADER);
         if (message == null) return false;
-        return message.getEnvelopeType() == WonMessageDirection.FROM_SYSTEM
-                && message.getSenderURI() != null && message.getSenderURI().equals(message.getReceiverURI());
+        if (message.getEnvelopeType() != WonMessageDirection.FROM_SYSTEM) return false;
+        if (message.getSenderURI() != null) {
+            return message.getSenderURI().equals(message.getReceiverURI());
+        } else {
+            if (message.getSenderNeedURI() == null) return false;
+            return message.getSenderNeedURI().equals(message.getReceiverNeedURI());
```

Comment:
looks to me that this line is never called

Reply:
I don't see that
It can be confusing to have the check for senderURI/receiverURI and senderNeedURI/receiverNeedURI - is that it?

Reply:
ahh, i didnt see the difference between "getSenderURI()" and "getSenderNeedURI()", ok